src/main/java/com/exactpro/th2/readcsv/Main.java

```diff
@@ -95,33 +102,66 @@ public static void main(String[] args) {
-
-            AbstractFileReader<LineNumberReader> reader = new DefaultFileReader.Builder<LineNumberReader>(
-                    configuration.getCommon(),
-                    directoryChecker,
-                    new CsvContentParser(configuration.getAliases()),
-                    new MovedFileTracker(configuration.getSourceDirectory()),
-                    new InMemoryReaderState(),
-                    streamId -> commonFactory.newMessageIDBuilder().build(),
-                    Main::createSource
-            )
-                    .readFileImmediately()
-                    .acceptNewerFiles()
-                    .onSourceFound((streamId, path) -> clearHeader(headerHolder, streamId))
-                    .onContentRead((streamId, path, builders) -> attachHeaderOrHold(headerHolder, streamId, builders, configuration))
-                    .onStreamData((streamId, builders) -> publishMessages(rawMessageBatchRouter, streamId, builders))
-                    .onError((streamId, message, ex) -> publishErrorEvent(eventBatchRouter, streamId, message, ex, rootId))
-                    .onSourceCorrupted((streamId, path, e) -> publishSourceCorruptedEvent(eventBatchRouter, path, streamId, e, rootId))
-                    .build();
-            toDispose.add(reader);
+            final Runnable readerProcessUpdates;
+
+            if (configuration.isUseTransport()) {
+                var headerHolder = new TransportHeaderHolder(configuration.getAliases());
+                var router = commonFactory.getTransportGroupBatchRouter();
+
+                AbstractFileReader<LineNumberReader, com.exactpro.th2.common.schema.message.impl.rabbitmq.transport.RawMessage.Builder, MessageId.Builder> reader = new TransportDefaultFileReader.Builder<>(
+                        configuration.getCommon(),
+                        directoryChecker,
+                        new TransportCsvContentParser(configuration.getAliases()),
+                        new MovedFileTracker(configuration.getSourceDirectory()),
+                        new InMemoryReaderState(),
+                        streamId -> MessageId.builder(),
+                        Main::createSource
+                )
+                        .readFileImmediately()
+                        .acceptNewerFiles()
+                        .onSourceFound((streamId, path) -> clearHeader(headerHolder, streamId))
+                        .onContentRead((streamId, path, builders) -> transportAttachHeaderOrHold(headerHolder, streamId, builders, configuration))
+                        .onStreamData((streamId, builders) -> publishTransportMessages(router, streamId, builders, boxBookName))
+                        .onError((streamId, message, ex) -> publishErrorEvent(eventBatchRouter, streamId, message, ex, rootId))
+                        .onSourceCorrupted((streamId, path, e) -> publishSourceCorruptedEvent(eventBatchRouter, path, streamId, e, rootId))
+                        .build();
```

Comment:
I think we must extract reader initialization into separate classes because right now the Main method looks like a mess... There are a lot of fully-qualified imports and mix of methods for proto and transport messages